### DISCUSSION TITLE
lightGallery and GPL-3.0

### DISCUSSION THREAD

**Discussion by HEIGE-PCloud** [HEIGE-PCloud/DoIt#317] [General]
Currently, the DoIt theme is using [lightGallery 1.x](https://github.com/sachinchoolur/lightgallery.js) to provide advance image gallery functions. Since the 1.x version is no longer maintained, I am consider migrating to v2.x, and I noticed that lightGallery uses the GPL-3.0 license in both [1.x](https://github.com/sachinchoolur/lightgallery.js/blob/master/LICENSE.md) and [2.x](https://github.com/sachinchoolur/lightGallery/blob/master/LICENSE). The DoIt theme is using the MIT License, for which I believe is not quite compatible with GPL-3.0.

I think it is not very appropriate to keep using lightGallery in this project, although no one has complaint about it yet. Any ideas about replacements for lightGallery?

**Comment by eyllanesc** [comment]
maybe https://github.com/dimsemenov/photoswipe

**Comment by HEIGE-PCloud** [comment]
The following dependencies all use GPL-3 license:

- [sharer.js](https://github.com/ellisonleao/sharer.js)
- [lightgallery.js](https://github.com/sachinchoolur/lightgallery.js)
- [TypeIt](https://github.com/alexmacarthur/typeit)

Removing and replacing them will be a low-priority task to do…

**Reply by HEIGE-PCloud** [reply]
Maybe [Typed.js](https://github.com/mattboldt/typed.js) can replace TypeIt.js…
